common/elasticsearch/bulk_processor_v6.go

```diff
@@ -0,0 +1,200 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package elasticsearch
+
+import (
+	elastic6 "github.com/olivere/elastic"
+	"github.com/olivere/elastic/v7"
+)
+
+type (
+	bulkProcessorV6 struct {
+		esBulkProcessor *elastic6.BulkProcessor
+	}
+)
+
+func newBulkProcessorV6(esBulkProcessor *elastic6.BulkProcessor) *bulkProcessorV6 {
+	if esBulkProcessor == nil {
+		return nil
+	}
```

Comment:
when will `esBulkProcessor == nil`?

Reply:
Whenever `c.esClient.BulkProcessor().Do(ctx)` returns it (I believe in case of error only). Essentially it is a wrapper and I want wrapper to be `nil` every time when underlying object is `nil`.

Reply:
> Whenever `c.esClient.BulkProcessor().Do(ctx)` returns it (I believe in case of error only). Essentially it is a wrapper and I want wrapper to be `nil` every time when underlying object is `nil`.

is that weird? wrapper over nil?

Reply:
It just simplify caller code w/o checking for `nil` and `err` there. I think it is ok.